### PR TITLE
Chapter 12: Reset Password

### PR DESCRIPTION
**Pull request by ainulfarizah**
ready

### CODE REVIEW

**Comment by ainulfarizah** [comment]
ready

File: app/controllers/password_resets_controller.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
+class PasswordResetsController < ApplicationController
+  before_action :get_user, :valid_user, :check_expiration, only: [:edit, :update], :check_expiration
```
Comment (ghost) [inline app/controllers/password_resets_controller.rb]:
```suggestion
  before_action :get_user, :valid_user, :check_expiration, only: [:edit, :update]
```

File: test/mailers/user_mailer_test.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+require "test_helper"
```
Comment (ghost) [inline test/mailers/user_mailer_test.rb]: delete this file please

File: test/mailers/previews/user_mailer_preview.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# Preview all emails at http://localhost:3000/rails/mailers/user_mailer
+class UserMailerPreview < ActionMailer::Preview
```
Comment (ghost) [inline test/mailers/previews/user_mailer_preview.rb]: delete this file please

**Review by ghost — COMMENTED** [review]
cmt

**Comment by ainulfarizah** [comment]
ready

File: app/helpers/account_activations_helper.rb [hunk 4]
```diff
@@ -0,0 +1,2 @@
+module AccountActivationsHelper
```
Comment (ngocvt-0484) [inline app/helpers/account_activations_helper.rb]: delete this file

File: config/routes.rb [hunk 5]
```diff
@@ -7,9 +7,12 @@
     post "/signup", to:"users#create"
     resources :users
     resources :account_activations, only: [:edit]
+    resources :password_resets, only: [:new, :create, :edit, :update]
 
     get "/login", to: "sessions#new"
     post "/login", to: "sessions#create"
     delete "/logout", to: "sessions#destroy"
+    get 'password_resets/new'
```
Comment (ngocvt-0484) [inline config/routes.rb]: remove it because you defined on line 10

File: config/routes.rb [hunk 6]
```diff
@@ -7,9 +7,12 @@
...
+    get 'password_resets/edit'
```
Comment (ngocvt-0484) [inline config/routes.rb]: remove it because you defined on line 10

**Comment by ainulfarizah** [comment]
ready

**Review by ghost — COMMENTED** [review]
merged

**Pull request merged**